src/stack.py

```diff
@@ -7,15 +7,17 @@ class Stack(object):
 
     def __init__(self, inbound_data=None):
         """Create new Stack composing from LinkedList."""
-        self.inbound_data = None
         self._linked_list = LinkedList(inbound_data)
+        self.head = self._linked_list.head
 
     def push(self, val):
         """Return a new node on the Stack using LinkedList.push."""
         self._linked_list.push(val)
+        self.head = self._linked_list.head
 
     def pop(self):
         """Return pop method for LinkedList on Stack."""
+        self.head = self._linked_list.head
         return self._linked_list.pop()
```

Comment:
Notice: you're reassigning the `head` before the pop, not after. So now your pointer will be pointing at the wrong thing.